src/request.rs

```diff
@@ -587,6 +676,32 @@ impl<'a> Request<'a> {
         self
     }
 
+    ///Sets connection timeout of request.
+    ///By default timeout is set to 3 hours.
```

Comment:
There should not be a default timeout for the entire request. That would unconditionally interrupt long downloads.

Reply:
Now timeout is optional (`Option<Duration>`). By default there is no timeout.